CIPs/cip-0032.md

```diff
@@ -13,81 +13,105 @@ license: Apache 2.0
-- `AttestationSlasherPenalty` Similarly the amount of stake slashed will start conservatively. At the time of writing this, only two active validators have an excess stake of less than 100 CELO (85 CELO each) while all other validators have more than this. If the penalty is 40 CELO, then these validators can be slashed twice before incurring any loss in epoch rewards. The penalty will start conservative with a minimum penalty of 5 CELO and increase the amount to 40 CELO once the community is confident in the slashing rules.
-- `AttestationSlasherReward` The reward for slashing should cover both the gas costs as well as the infrastructure costs to monitor for offenses.
+### Defining parameters
 
-#### MinStakeSlasher
+- `TargetCompletionRate` - From a week's worth of recent data (Feb 5th, 2021) a vast majority of Attestation Services perform above 70% completion rate. Only 7/97 services performed below 70%, all which were above 60%. The proposed starting point for the target is `.7` and can be raised if other areas of the protocol reliability are improved that would naturally raise the completion rate of all services.
+- `AttestationServiceRewardPercentage` - Setting `.20` would allocate 15k cUSD/year to run an Attestation Service and 60 cUSD/year to run a Validator Service.
+- `MinGlobalAttestationCompletionRate` - The global completion rate rarely dips below 70%. We can set this safety net at `.5` and adjust it lower if it was ever necessary.
+- `GlobalAttestationFailureFallbackReward` - The starting point will be `.75`. This is low enough to encourage Attestation Services to come back online in the case of a total outage, but high enough to not significantly impact rewards.
 
-- `MinStakeSlasherPenalty` The penalty is only the slasher reward (no additional penalty) since the validator has already been slashed to get to this point.
-- `MinStakeSlasherReward` Similarly, the reward for slashing should cover both the gas costs as well as the infrastructure costs to monitor for offenses.
+- `AttestationMinPeriod` - An Attestation Service should complete at least 1 attestation over 2 days. This number can be more aggressive since there's no penalty and the validator can always re-register to get back in.
+- `AttestationMinRequested` - 8.
+- `MinNumberOfAttestationIssuers` - 40. Any lower than this and the random selection of issuers is too easy to game.
```

Comment:
Do we have anywhere the attack model for this? It may be worth adding that here is nowhere else. I'd be happy to write up such an analysis.

Reply:
I haven't seen any model for this, but it's a good idea to call it out explicitly. If you're interested in adding it, it'd be much appreciated 🙏 It'd be interesting to see the expected cost to "hijack" a phone number with different numbers of colluders.

Reply:
On the topic of hijacking a phone number, I include some analysis in https://github.com/celo-org/celo-monorepo/issues/7057#issuecomment-779639780